src/componententity.cpp

```diff
@@ -24,47 +24,36 @@ limitations under the License.
 #include "libcellml/model.h"
 #include "libcellml/units.h"
 
+#include "component_p.h"
 #include "utilities.h"
 
 namespace libcellml {
 
-/**
- * @brief The ComponentEntity::ComponentEntityImpl struct.
- *
- * This struct is the private implementation struct for the ComponentEntity class.  Separating
- * the implementation from the definition allows for greater flexibility when
- * distributing the code.
- */
-struct ComponentEntity::ComponentEntityImpl
-{
-    std::vector<ComponentPtr> mComponents;
-    std::string mEncapsulationId;
-
-    std::vector<ComponentPtr>::iterator findComponent(const std::string &name);
-    std::vector<ComponentPtr>::iterator findComponent(const ComponentPtr &component);
-};
-
-std::vector<ComponentPtr>::iterator ComponentEntity::ComponentEntityImpl::findComponent(const std::string &name)
+std::vector<ComponentPtr>::const_iterator ComponentEntity::ComponentEntityImpl::findComponent(const std::string &name) const
 {
     return std::find_if(mComponents.begin(), mComponents.end(),
                         [=](const ComponentPtr &c) -> bool { return c->name() == name; });
 }
 
-std::vector<ComponentPtr>::iterator ComponentEntity::ComponentEntityImpl::findComponent(const ComponentPtr &component)
+std::vector<ComponentPtr>::const_iterator ComponentEntity::ComponentEntityImpl::findComponent(const ComponentPtr &component) const
 {
     return std::find_if(mComponents.begin(), mComponents.end(),
                         [=](const ComponentPtr &c) -> bool { return c->equals(component); });
 }
 
-// Interface class Model implementation
-ComponentEntity::ComponentEntity()
-    : mPimpl(new ComponentEntityImpl())
+inline ComponentEntity::ComponentEntityImpl *ComponentEntity::pFunc()
```

Comment:
Don't bother with `inline` (compilers are clever enough to do the right thing nowadays), not least since we should otherwise `inline` quite a few other "trivial" methods too.

Reply:
Some aren't, with the mix of building on Windows, macOS, and Ubuntu with building using SWIG and Emscripten the `inline` keyword becomes important.

Reply:
We might then want to review our current use of `inline` since we, for instance, use it with some instances of `pFunc()` but not some others.

Reply:
a7f9eb8